gamemode/modules/base/sh_gamemode_functions.lua

```diff
@@ -8,7 +8,7 @@ end
 function GM:StartCommand(ply, usrcmd)
     -- Used in arrest_stick and unarrest_stick but addons can use it too!
     local wep = ply:GetActiveWeapon()
-    if IsValid(wep) and isfunction(wep.startDarkRPCommand) then
+    if wep:IsValid() and isfunction(wep.startDarkRPCommand) then
```

Comment:
Even though `ply:GetActiveWeapon()` returns `NULL` when the player has no weapons, I prefer `IsValid` in principle, because that allows me to ignore the difference between `nil` and `NULL`. 

Usage of `ent:IsValid()` raises the question *"Oh, but can this thing be `nil`?"*, and I have to check the wiki and/or the implementation of the function to see if this is the case. I argue that readability here is more important than the unnoticeable performance benefit.

Reply:
We already changed it in a lot f places.
I just want it to be changed in hooks being called very often.
The IsValid check in general here isn't needed I think.

Reply:
The GMod API has slowly been moving toward NULL consistency - there are only a few small edge cases left (NPC:GetEnemy and GetTarget specifically) that are waiting to be resolved. Differentiating between nil and NULL is now an almost completely safe paradigm that is well-defined and yields constant time optimisation.